experimental/LieAlgebras/src/simple_lie_algebra.jl

```diff
@@ -0,0 +1,174 @@
+#Construct a simple Lie algebra over a given field with a given root system
+@attributes mutable struct SimpleLieAlgebra{C<:FieldElem} <: LieAlgebra{C}
+  R::Field
+  root_system::RootSystem
+  dim::Int
+  s::Vector{Symbol}
+  root_system_type::Tuple{Symbol,Int64}
+  struct_consts::Matrix{SRow{C}}
+  function SimpleLieAlgebra{C}(
+    R::Field, S::Symbol, n::Int64; cached::Bool=true
+  ) where {C<:FieldElem}
+    RS = root_system(S, n)
+    dimL = number_of_roots(RS) + length(RS.simple_roots)
+    s = [Symbol("e_$i") for i in 1:dimL]
+    st = root_system_type(RS)
+    #get the structure constants of the Lie algebra L 
+    #note that it is enough to do this over QQ, as we can later coerce the constants
+    #into the field R
+    coeffs_iso = inv(Oscar.iso_oscar_gap(QQ))
+    LG = GAP.Globals.SimpleLieAlgebra(GAP.Obj(S), n, domain(coeffs_iso))
+    sc_table_G =
+      (
+        entry -> (entry[1], Vector{elem_type(QQ)}(map(coeffs_iso, entry[2])))
+      ).(
+        Matrix{Tuple{Vector{Int},Vector{GAP.Obj}}}(
+          (GAP.Globals.StructureConstantsTable(GAPWrap.Basis(LG)))[1:dimL]
+        )
+      )
+    struct_consts = Matrix{SRow{elem_type(R)}}(undef, dimL, dimL)
+    for i in 1:dimL, j in 1:dimL
+      struct_consts[i, j] = sparse_row(
+        R, Tuple{Int,elem_type(R)}[(k, R(c)) for (k, c) in zip(sc_table_G[i, j]...)]
+      )
+    end
+    return get_cached!(SimpleLieAlgebraDict, (R, RS), cached) do
+      new{C}(R, RS, dimL, s, st, struct_consts)
+    end::SimpleLieAlgebra{C}
+  end
+end
+
+const SimpleLieAlgebraDict = CacheDictType{Tuple{Field,RootSystem},SimpleLieAlgebra}()
+
+mutable struct SimpleLieAlgebraElem{C<:FieldElem} <: LieAlgebraElem{C}
+  parent::SimpleLieAlgebra{C}
+  mat::MatElem{C}
+end
+
+###############################################################################
+#
+#   Basic manipulation
+#
+###############################################################################
+
+parent_type(::Type{SimpleLieAlgebraElem{C}}) where {C<:FieldElem} = SimpleLieAlgebra{C}
+
+elem_type(::Type{SimpleLieAlgebra{C}}) where {C<:FieldElem} = SimpleLieAlgebraElem{C}
+
+parent(x::SimpleLieAlgebraElem) = x.parent
+
+base_ring(L::SimpleLieAlgebra{C}) where {C<:FieldElem} = L.R::parent_type(C)
+
+coefficient_ring(L::SimpleLieAlgebra{C}) where {C<:FieldElem} = L.R::parent_type(C)
+
+dim(L::SimpleLieAlgebra) = L.dim
+
+root_system(L::SimpleLieAlgebra) = L.root_system
+
+root_system_type(L::SimpleLieAlgebra) = L.root_system_type
+
+#  Base.@propagate_inbounds function Base.setindex!(f::SimpleLieAlgebraElem{T}, d, r) where {T <: FieldElem}
+#    Generic._matrix(f)[1, r] = d
+#    return f
+#  end
+
```

Comment:
```suggestion
```
As already explained earlier, this can break a lot of stuff. And you have no speed improvement over first fiddling around with your coefficient vector and then, in the end, wrapping it into a LieAlgebraElem. And, if we decide to have something like this in the future, it should be available for the abstract supertype.

But not a blocker for this PR, as it is only a comment.